sql/core/src/main/scala/org/apache/spark/sql/Observation.scala

```diff
@@ -150,10 +150,12 @@ class Observation(name: String) {
 private[sql] case class ObservationListener(observation: Observation)
   extends QueryExecutionListener {
 
-  override def onSuccess(funcName: String, qe: QueryExecution, durationNs: Long): Unit =
+  override def onSuccess(
+      funcName: String, executionId: Long, qe: QueryExecution, durationNs: Long): Unit =
```

Comment:
can't we get the id by `qe.id`?

Reply:
oh they are not the same thing. We should probably unify these two

Reply:
Yes, they are not the same. Definitely, it will great if we can unify these.

Reply:
Checked the original motivation of the queryExecution.id in this PR https://github.com/apache/spark/pull/29372. Maybe take query execution id also works for it.
Then maybe we can lazily set a copy of query execution id to QeuryExecution object when [SQLExecution.withNewExecutionId](https://github.com/apache/spark/blob/v3.2.0-rc2/sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala#L62) is invoked and query execution id is generated.

Reply:
one idea is to remove `SQLExecution.nextExecutionId` and always use `QueryExecution.id`

Reply:
If use QueryExecution.id, the queryId in SQL page of UI may not be sequential if I got the right understanding. If it is the case, would that be a problem?

Reply:
I see, can we make `QueryExecution.id` sequential?

Reply:
If we want to unify these two, I think the id bound to a QueryExecution should be the same with query execution id we can get from UI now. So the timing to initialize the id might also need to be the same.

Since a QueryExecution will be created for each dataset, we can not initialize the id for every QueryExecution, but  just for the ones triggered by an action. I think the logic and timing here is just like what we did in SQLExecution.nextExecutionId.
In another word, we can just put the query execution Id to a QueryExecution when newExecutionId generated. This is what I have in mind now.

Reply:
But this also has some problems, since a `QueryExecution`/`Dataset` can be reused for query execution multi times, so we may get same `QueryExecution` with different execution ids.